### PR TITLE
Added a bit more support for the SenseHat.

### PR DESCRIPTION
**Pull request by Walter-Stroebel**
Added support for the LPS25H on the SenseHat.

### CODE REVIEW

**Comment by Walter-Stroebel** [comment]
I signed it!

**Comment by proppy** [comment]
Related #24

File: sensehat/README.md [hunk 1]
```diff
@@ -54,6 +54,17 @@ display.draw(bitmap);
 // Close the display when done.
 display.close();
 ```
+```
+                try (BaroTemp bt = SenseHat.openBaroTemp()) {
```
Comment (Fleker) [inline sensehat/README.md]: I think you're indenting too much here.
Your snippet also lacks a comment of what it does.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+import java.io.IOException;
+
+/**
+ * This class allows access to the LPS25H on the SenseHat.
+ * <p>
+ * See also: https://www.pololu.com/file/download/LPS25H.pdf?file_id=0J761</p>
+ * <p>Source code referenced: https://github.com/tkurbad/mipSIE/blob/master/python/AltIMU-10v5/i2c.py</p>
+ */
+public class BaroTemp implements AutoCloseable {
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Generally our class names closely reflect the actual name of the part, this being the LPS25H. Then the comments in the class would describe that it is a temperature/barometric pressure sensor.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    private final int LPS_FIFO_CTRL = 0x2E; //  FIFO control / mode selection
+    private final int LPS_FIFO_STATUS = 0x2F; //  FIFO status
+
+    private final int LPS_THS_P_L = 0x30; //  Pressure interrupt threshold, low byte
+    private final int LPS_THS_P_H = 0x31; //  Pressure interrupt threshold, high byte
+
+    // The next two registers need special soldering
+    private final int LPS_RPDS_L = 0x39;//  Pressure offset for differential pressure computing, low byte
+    private final int LPS_RPDS_H = 0x3A; //  Differential offset, high byte
+    private int milliBarAdjust = 0;
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Nit, code style for fields are `m<Name>`

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        ret += (mDevice.readRegByte(a1) & 0xFF) << 8;
+        if (ret < 32768) return ret;
+        else return ret - 65536;
+    }
+
+    /**
+     * The sensor seems to have an offset to the actual pressure. You can find your local "real" pressure quite easily on the web. Get the measured value from the
+     * sensor and compute the difference. The value obtained can be passed to this method to "calibrate" your board's sensor.
+     *
+     * @param hPa difference to actual air pressure
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Can you mention the units?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+    private int readSigned16(int a0, int a1) throws IOException {
+        int ret = (mDevice.readRegByte(a0) & 0xFF);
+        ret += (mDevice.readRegByte(a1) & 0xFF) << 8;
+        if (ret < 32768) return ret;
+        else return ret - 65536;
+    }
+
+    /**
+     * The sensor seems to have an offset to the actual pressure. You can find your local "real" pressure quite easily on the web. Get the measured value from the
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Code style is that the length of each line be no more than 100 chars. Can you review your line lengths?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+     */
+    public double getBarometer() throws IOException {
+        return getBarometerRaw() / 4096.0;
+    }
+
+    /**
+     * Fetch the temperature in degrees Celcius. Note that the design of the SenseHat makes this more the
+     * temperature of the board then the actual (room) temperature!
+     *
+     * @return The temperature as reported by the sensor.
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Can you specify the units here as well?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+                Thread.sleep(1);
+            } catch (InterruptedException e) {
+                throw new IOException(e);
+            }
+        }
+        return readSigned16(LPS_TEMP_OUT_L, LPS_TEMP_OUT_H);
+    }
+
+    /**
+     * Fetch air pressure in hPa (milli bar). Note that this call waits for data to be available.
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: I believe it's just millibar, without a space.

Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: How long do we expect to wait? Does it make sense to make this asynchronous?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java [hunk 8]
```diff
@@ -23,12 +23,17 @@
  */
 @SuppressWarnings({"unused", "WeakerAccess"})
 public class SenseHat {
-    public static final int I2C_ADDRESS =  0x46;
-    public static final String BUS_DISPLAY = "I2C1";
+    public static final int I2C_DISPLAY_ADDRESS =  0x46;
+    public static final int I2C_LPS25H_ADDRESS =  0x46;
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: They're the same address?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java [hunk 9]
```diff
@@ -23,12 +23,17 @@
...
+    public static final String BUS_NAME = "I2C1";
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: We should make this bus name a parameter that can be provided, for a variety of devices with different bus names.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java [hunk 10]
```diff
@@ -23,12 +23,17 @@
...
     public static final int DISPLAY_WIDTH = LedMatrix.WIDTH;
     public static final int DISPLAY_HEIGHT = LedMatrix.HEIGHT;
 
     public static LedMatrix openDisplay() throws IOException {
-        return new LedMatrix(BUS_DISPLAY);
+        return new LedMatrix(BUS_NAME);
+    }
+
+    public static BaroTemp openBaroTemp() throws IOException {
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: Can you add JavaDoc for this method?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java [hunk 11]
```diff
@@ -23,12 +23,17 @@
...
 public class SenseHat {
-    public static final int I2C_ADDRESS =  0x46;
-    public static final String BUS_DISPLAY = "I2C1";
+    public static final int I2C_DISPLAY_ADDRESS =  0x46;
+    public static final int I2C_LPS25H_ADDRESS =  0x46;
+    public static final String BUS_NAME = "I2C1";
     public static final int DISPLAY_WIDTH = LedMatrix.WIDTH;
     public static final int DISPLAY_HEIGHT = LedMatrix.HEIGHT;
 
     public static LedMatrix openDisplay() throws IOException {
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: Can you add Javadoc for this method as well?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+     *
+     * @param hPa difference to actual air pressure
+     * @throws IOException from I2cDevice
+     */
+    public void setBarometerOffset(double hPa) throws IOException {
+        this.milliBarAdjust = (int) Math.round(hPa * 4096);
+    }
+
+    /**
+     * Fetch raw value, see the data sheet. Note that this call waits for data to be available.
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Can you link to the datasheet in this comment?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+                Thread.sleep(1);
+            } catch (InterruptedException e) {
+                throw new IOException(e);
+            }
+        }
+        return readSigned24(LPS_PRESS_OUT_XL, LPS_PRESS_OUT_L, LPS_PRESS_OUT_H) + milliBarAdjust;
+    }
+
+    /**
+     * Fetch raw value, see the data sheet. Note that this call waits for data to be available.
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Can you link to the datasheet in this comment?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 14]
```diff
@@ -0,0 +1,170 @@
+package com.google.android.things.contrib.driver.sensehat;
+
+import com.google.android.things.pio.I2cDevice;
+import com.google.android.things.pio.PeripheralManager;
+
+import java.io.IOException;
+
+/**
+ * This class allows access to the LPS25H on the SenseHat.
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Generally with our sensors we have two classes. One implements the raw protocol, `Lps25h`, and the other plugs into the sensor framework, ie. `Lps25hSensorDriver`.

One example is in our [BMX280](https://github.com/androidthings/contrib-drivers/tree/master/bmx280/src/main/java/com/google/android/things/contrib/driver/bmx280)

File: sensehat/src/androidTest/java/com/google/android/things/contrib/driver/sensehat/SenseHatDeviceTest.java [hunk 15]
```diff
@@ -38,34 +39,38 @@
     @Test
     public void senseHat_DisplayColor() throws IOException {
         // Color the LED matrix.
-        LedMatrix display = SenseHat.openDisplay();
-
-        display.draw(Color.MAGENTA);
-        // Close the display when done.
-        display.close();
+        try (LedMatrix display = SenseHat.openDisplay()) {
```
Comment (Fleker) [inline sensehat/src/androidTest/java/com/google/android/things/contrib/driver/sensehat/SenseHatDeviceTest.java]: Should we be doing `try` here? If it doesn't work, I think we'd want the test to fail.

File: sensehat/README.md (hunk 1 above)
Comment (Walter-Stroebel) [reply sensehat/README.md]: Fixed. And thanks for the awesome code review!

File: sensehat/src/androidTest/java/com/google/android/things/contrib/driver/sensehat/SenseHatDeviceTest.java (hunk 15 above)
Comment (Walter-Stroebel) [reply sensehat/src/androidTest/java/com/google/android/things/contrib/driver/sensehat/SenseHatDeviceTest.java]: It will, there is no catch(). I changed this to use the autoclose feature which was implemented but never used / demonstrated by the original author.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java (hunk 5 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Fixed.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java (hunk 4 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Done.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java (hunk 12 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: There is a global reference to the datasheet for the entire class.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java (hunk 7 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Fixed.

Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Fixed in description. Usually the wait will be just the overhead for fetching the status register as it makes little sense for an application to poll the device more than 12.5 times a second so data should always be available. The JavaDoc now explains this.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java (hunk 6 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Done

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java (hunk 8 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: Good catch! Fixed!

**Comment by MacroYau** [comment]
Hey @proppy, what is the current plan for the Sense HAT drivers?

Is there anything I can help further #24 (and possibly #86 which fills the gap of my PR)?

**Comment by proppy** [comment]
@MacroYau Yep, sorry I never got to send an update update to your PR about the SenseHat, I'd love to make progress on this and I'm happy to help with the review.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java (hunk 9 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: Yes we should! However, this is from the original code and I daresay a common issue through the other libraries as well. Maybe there should be a global device data object to address this?
Meanwhile I've provided overloaded factory methods.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java (hunk 11 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: Done.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java (hunk 10 above)
Comment (Walter-Stroebel) [reply sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: Done.

**Comment by Walter-Stroebel** [comment]
@MacroYau @proppy 
I'm sorry, I was not aware of #24 
Unfortunately time constraints prohibit me from merging the two related PR's at the moment.
Feel free to either decline my PR, merge it with #24 or merge #24 with my PR, at your convenience.

File: sensehat/README.md [hunk 16]
```diff
@@ -4,7 +4,7 @@ Sense Hat driver for Android Things
 This driver provides easy access to the peripherals available on the Raspberry Pi [Sense Hat][product]:
 - 8x8 LED matrix
 - TODO: 5 buttons joystick
-- TODO: Sensors
+- TODO: Sensors other then LPS25H
```
Comment (Fleker) [inline sensehat/README.md]: I'd rephrase this list:

- 8x8 LED matrix
- LPS25H
...
- TODO: Other sensors

File: sensehat/README.md [hunk 17]
```diff
@@ -54,6 +54,19 @@ display.draw(bitmap);
 // Close the display when done.
 display.close();
 ```
+```
```
Comment (Fleker) [inline sensehat/README.md]: You can add Java after the backticks to add code highlighting to the code block. Also, you won't need to add additional indenting.

```java
// just a snippet that verifies the device is working
// your actual use would of course be much more awesome
try (BaroTemp bt = SenseHat.openBaroTemp()) {
    bt.setBarometerOffset(-6.2);
    Log.i(TAG, "PressureRaw: " + bt.getBarometerRaw());
    Log.i(TAG, "TemperatureRaw: " + bt.getTemperatureRaw());
    Log.i(TAG, "Pressure: " + bt.getBarometer());
    Log.i(TAG, "Temperature: " + bt.getTemperature());
} catch (Exception e) {
    Log.e(TAG, "Oops", e);
}
```

File: sensehat/README.md [hunk 18]
```diff
@@ -54,6 +54,19 @@ display.draw(bitmap);
...
+    // just a snippet that verifies the device is working
```
Comment (Fleker) [inline sensehat/README.md]: I'm not sure if we need the comments, as the snippet is fairly descriptive.

File: sensehat/src/androidTest/java/com/google/android/things/contrib/driver/sensehat/SenseHatDeviceTest.java (hunk 15 above)
Comment (Fleker) [reply sensehat/src/androidTest/java/com/google/android/things/contrib/driver/sensehat/SenseHatDeviceTest.java]: So what happens if `SenseHat.openDisplay()` fails? Does the test still pass because it's within a `try` block?

File: sensehat/src/androidTest/java/com/google/android/things/contrib/driver/sensehat/SenseHatDeviceTest.java [hunk 19]
```diff
@@ -38,34 +39,38 @@
...
+            paint.setShader(new RadialGradient(4, 4, 4, Color.RED, Color.BLUE, Shader.TileMode.CLAMP));
+            canvas.drawRect(0, 0, canvas.getWidth(), canvas.getHeight(), paint);
+            display.draw(bitmap);
+        }
+    }
+
+    @Test
+    public void senseHat_BaroTemp() throws IOException {
+        try (BaroTemp baroTemp = SenseHat.openBaroTemp()) {
+            Log.i(BaroTemp.class.getName(),"SenseHat Barotemp raw pressure: "+baroTemp.getBarometerRaw());
```
Comment (Fleker) [inline sensehat/src/androidTest/java/com/google/android/things/contrib/driver/sensehat/SenseHatDeviceTest.java]: Is there any way we can add assertions to this? Can we verify that the values are non-zero?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 20]
```diff
@@ -0,0 +1,190 @@
...
+ * <p>Source code referenced: https://github.com/tkurbad/mipSIE/blob/master/python/AltIMU-10v5/i2c.py</p>
+ */
+public class BaroTemp implements AutoCloseable {
+    private I2cDevice mDevice;
+    // Register addresses
+    private final int LPS_REF_P_XL = 0x08; //  Reference pressure, lowest byte
+    private final int LPS_REF_P_L = 0x09; //  Reference pressure, low byte
+    private final int LPS_REF_P_H = 0x0A; //  Reference pressure, high byte
+
+    private final int LPS_WHO_AM_I = 0x0F; //  Returns 0xbd (read only)
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: I'm not sure what this value is supposed to represent?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 21]
```diff
@@ -0,0 +1,190 @@
...
+     * Create a new barometric pressure and temperature sensor driver connected on the given I2C bus.
+     *
+     * @param bus I2C bus the sensor is connected to.
+     * @throws IOException when a lower level does
+     */
+    public BaroTemp(String bus) throws IOException {
+        PeripheralManager pioService = PeripheralManager.getInstance();
+        mDevice = pioService.openI2cDevice(bus, SenseHat.I2C_LPS25H_ADDRESS);
+        // power down first
+        mDevice.writeRegByte(LPS_CTRL_REG1, (byte) 0);
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: Can we assume the device is in a powered-down state?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 22]
```diff
@@ -0,0 +1,190 @@
...
+                mDevice = null;
+            }
+        }
+    }
+
+    private int readSigned24(int a0, int a1, int a2) throws IOException {
+        int ret = (mDevice.readRegByte(a0) & 0xFF);
+        ret += ((int) mDevice.readRegByte(a1) & 0xFF) << 8;
+        ret += ((int) mDevice.readRegByte(a2) & 0xFF) << 16;
+        if (ret < 8388608) return ret;
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: [Google's style guide](https://google.github.io/styleguide/javaguide.html#s4.1-braces) states that conditionals should be wrapped in curly braces, even if it's one line,

ie. 
```java
if (ret < 8388608) {
    return ret;
}
```

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 23]
```diff
@@ -0,0 +1,190 @@
...
+     * pressure quite easily on the web. Get the measured value from the sensor and compute the
+     * difference. The value obtained can be passed to this method to "calibrate" your board's
+     * sensor. In the author's case the difference was 6.2 hPa which is quite significant. This
+     * error was confirmed with another (unrelated) sensor.
+     *
+     * @param hPa difference to actual air pressure in hectoPascal (hPa) or millibar.
+     * @throws IOException from I2cDevice
+     */
+    public void setBarometerOffset(double hPa) throws IOException {
+        this.mMillibarAdjust = (int) Math.round(hPa * 4096);
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: The use of `this` is not necessary as the `m` prefix to the variable name implies it is part of the class.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 24]
```diff
@@ -0,0 +1,190 @@
...
+     * The sensor seems to have an offset to the actual pressure. You can find your local "real"
+     * pressure quite easily on the web. Get the measured value from the sensor and compute the
+     * difference. The value obtained can be passed to this method to "calibrate" your board's
+     * sensor. In the author's case the difference was 6.2 hPa which is quite significant. This
+     * error was confirmed with another (unrelated) sensor.
+     *
+     * @param hPa difference to actual air pressure in hectoPascal (hPa) or millibar.
+     * @throws IOException from I2cDevice
+     */
+    public void setBarometerOffset(double hPa) throws IOException {
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: I don't think this method does any I2C writing, so it should not need to throw IOException.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 25]
```diff
@@ -0,0 +1,190 @@
...
+     * second there will be no wait.</p>
+     *
+     * @return The raw sensor value, adjusted by the given offset (if any).
+     * @throws IOException from I2cDevice
+     */
+    public int getBarometerRaw() throws IOException {
+        // wait for data available
+        while (0 == (mDevice.readRegByte(LPS_STATUS_REG) & 2)) {
+            try {
+                Thread.sleep(1);
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: I'm not sure if we should sleep a thread in trying to read a value. What happens if we remove it? Will the data just not be refreshed?

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java [hunk 26]
```diff
@@ -0,0 +1,190 @@
...
+     * Fetch raw value, see the data sheet. <p>Note that this call waits for data to be available.
+     * From the data sheet the (selected) refresh rate is 12.5 Hz so the max wait could be
+     * 1000/12.5 = 80 milliseconds with an average of 40 milliseconds. Call from asynchronous code
+     * if this is an issue. If your code calls this method less frequently then 12.5 times per
+     * second there will be no wait.</p>
+     *
+     * @return The raw sensor value, adjusted by the given offset (if any).
+     * @throws IOException from I2cDevice
+     */
+    public int getBarometerRaw() throws IOException {
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/BaroTemp.java]: In general, do we need to make these raw readings `public`? It doesn't seem like they'd be very useful to have.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java [hunk 27]
```diff
@@ -23,12 +23,47 @@
  */
 @SuppressWarnings({"unused", "WeakerAccess"})
 public class SenseHat {
-    public static final int I2C_ADDRESS =  0x46;
-    public static final String BUS_DISPLAY = "I2C1";
+    public static final int I2C_DISPLAY_ADDRESS =  0x46;
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: We probably should not rename existing public constants as that will break the API for developers.

File: sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java [hunk 28]
```diff
@@ -23,12 +23,47 @@
...
+        return new BaroTemp(BUS_NAME);
+    }
+
+    /**
+     * Connect to the LedMatrix of the SenseHat.
+     * @param busName The name of the I2C bus device on a non-raspi3 board.
+     * @return The autoclosing LedMatrix object.
+     * @throws IOException If caused by the hardware.
+     */
+    public static LedMatrix openDisplay(final String busName) throws IOException {
```
Comment (Fleker) [inline sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java]: Do these need to be final?
